### PR TITLE
Remove gulp and upgrade webpack

### LINKED ISSUE #2689: Replace Gulp with something more updated, gulp relies on to many outdated packages
Gulp has not been updated in three years and in nom land that is an eternity. I think we should replace it to clear out a lot of old dependencies and minor security issues.

Gulp is used to:

* Build CSS and JS for production and CI.
* Watch CSS and JS during development.
* Build React app via Webpack
* Analyse React app via Webpack
* Copy fonts and images for production.
* Run Djangos collectstatic.

All source files are in `/hypha/static_src/` and Gulp put the resulting files in `/hypha/static_compiled/`.

Djangos `collectstatic` then looks in `/hypha/static_compiled/` since it is added to `STATICFILES_DIRS`.

### LINKED ISSUE #2699: Update webpack, webpack-dev-server etc. to latest versions

### PR DESCRIPTION
**Pull request by vimal1083**
Fixes #2699
Fixes #2689

Add description here.

### CODE REVIEW

File: docker/Dockerfile.dev [hunk 1]
```diff
@@ -17,7 +17,7 @@ COPY package.json package-lock.json /usr/local/hypha/
 RUN sudo chown -R circleci:circleci /usr/local
 
 # Install node dependencies.
-RUN npm install  --quiet --global gulp-cli
+RUN npm install  --quiet --global
```
Comment (frjo) [inline docker/Dockerfile.dev]: This line can be removed since it was only for installing `gulp-cli`

**Comment by vimal1083** [comment]
Having some issues in upgrading webpack dev server and saas-loader. working on it

File: docker/Dockerfile.dev (hunk 1 above)
Comment (vimal1083) [reply docker/Dockerfile.dev]: sure will update

**Comment by vimal1083** [comment]
- For production deployment, we continue to use `npm run heroku-postbuild`.
- Backend Developer, who don't update react code while development can run `npm run dev:build` once (similar to production command, runs and finishes immediately. does not watch javascript files)
- Front end developers can use `npm run watch`, it will watch for javascript changes

cc: @frjo

**Comment by frjo** [comment]
Nice work!

We are missing some pieces however:

1. A watch command with sourcemaps and linting for CSS. This makes CSS development so much easier.
2. The watch command only watch the react app files. The apply/public javascript files do get rebuilt but only after a react app file has been touched.
3. The JavaScript production build is missing the "babel" and "uglify" steps. Or are you arguing that they are not needed? I'm always happy to remove stuff when possible.

**Review by sandeepsajan0 — APPROVED** [review]
Worked well for me as a backend developer with `npm run dev:build` and `npm run watch` commands.

**Comment by vimal1083** [comment]
1. A watch command with sourcemaps and linting for CSS. This makes CSS development so much easier.
    - Added source maps for CSS in dev build
3. The watch command only watch the react app files. The apply/public javascript files do get rebuilt but only after a react app file has been touched.
   - Added now, but HMR will not recognize the changes as these files are not part of react app, so the user has to refresh the browser to see the changes. I guess we had the same problem while using gulp.
5. The JavaScript production build is missing the "babel" and "uglify" steps. Or are you arguing that they are not needed? I'm always happy to remove stuff when possible.
   - We still use 'babel' in our webpack files, it helps us to transpile the ES6 javascript. Added TerserPlugin in webpack production file, it will mangle and compress the final Js output. We can have it. It adds an additional level of difficulty for others to read the javascript source code from prod build file.
 
cc: @frjo

**Comment by frjo** [comment]
@vimal1083 What do you think about the additions I have made to this PR?

1. Used "concurrently" and "nodemon" to implement watch commands for css and collectstatic.
2. Removed "babel-core", "node-sass-import-once" and "webpack-stream" since it seems they are not used anymore?
3. Updated "babel-jest" and "@svgr/webpack" to make "npm audit" a bit happier. Down to 5 critical.
4. Removed the "TerserPlugin", we do not need to obfuscate the js, this is open source.

Do you see any issues or things that need improvements?

**Comment by vimal1083** [comment]
I have seen your changes all of them looking good to me. Sorry for the delay

**Pull request merged**